pkg/controller/cluster/main-controller.go

```diff
@@ -861,97 +974,46 @@ func (c *Controller) syncHandler(key string) error {
 		// get a pod for the established statefulset
-		if tenant.Status.Pools[pi].State == miniov2.PoolCreated {
-			// get the first pod for the ss and try to reach the pool via that single pod.
-			pods, err := c.kubeClientSet.CoreV1().Pods(tenant.Namespace).List(ctx, metav1.ListOptions{
-				LabelSelector: fmt.Sprintf("%s=%s", miniov2.PoolLabel, pool.Name),
-			})
-			if err != nil {
-				klog.Warning("Could not validate state of statefulset for pool", err)
-			}
-			if len(pods.Items) > 0 {
-				var ssPod *corev1.Pod
-				for _, p := range pods.Items {
-					if strings.HasSuffix(p.Name, "-0") {
-						ssPod = &p
-						break
-					}
-				}
-				if ssPod == nil {
-					break
-				}
-				podAddress := fmt.Sprintf("%s:9000", tenant.MinIOHLPodHostname(ssPod.Name))
-				podAdminClnt, err := tenant.NewMinIOAdminForAddress(podAddress, tenantConfiguration, c.getTransport())
-				if err != nil {
-					return err
-				}
-
-				_, err = podAdminClnt.ServerInfo(ctx)
-				// any error means we are not ready, if the call succeeds or we get `server not initialized`, the ss is ready
-				if err == nil || madmin.ToErrorResponse(err).Code == "XMinioServerNotInitialized" {
-
-					// Restart the services to fetch the new args, ignore any error.
-					// only perform `restart()` of server deployment when we are truly
-					// expanding an existing deployment. (a pool became initialized)
-					minioRestarted := false
-					if len(tenant.Spec.Pools) > 1 && addingNewPool {
-						// get a new admin client that points to a pod of an already initialized pool (ie: pool-0)
-						livePods, err := c.kubeClientSet.CoreV1().Pods(tenant.Namespace).List(ctx, metav1.ListOptions{
-							LabelSelector: fmt.Sprintf("%s=%s", miniov2.PoolLabel, tenant.Spec.Pools[0].Name),
-						})
-						if err != nil {
-							klog.Warning("Could not validate state of statefulset for pool", err)
-						}
-						var livePod *corev1.Pod
-						for _, p := range livePods.Items {
-							if p.Status.Phase == v1.PodRunning {
-								livePod = &p
-								break
-							}
-						}
-						if livePod == nil {
-							break
-						}
-						livePodAddress := fmt.Sprintf("%s:9000", tenant.MinIOHLPodHostname(livePod.Name))
-						livePodAdminClnt, err := tenant.NewMinIOAdminForAddress(livePodAddress, tenantConfiguration, c.getTransport())
-						if err != nil {
-							return err
-						}
-						// Now tell MinIO to restart
-						if err = livePodAdminClnt.ServiceRestart(ctx); err != nil {
-							klog.Infof("We failed to restart MinIO to adopt the new pool: %v", err)
-						}
-						minioRestarted = true
-						metaNowTime := metav1.Now()
-						tenant.Status.WaitingOnReady = &metaNowTime
-						tenant.Status.CurrentState = StatusRestartingMinIO
-						if tenant, err = c.updatePoolStatus(ctx, tenant); err != nil {
-							klog.Infof("'%s' Can't update tenant status: %v", key, err)
-						}
-						klog.Infof("'%s' Was restarted", key)
-					}
+		if tenant.Status.Pools[pi].State == miniov2.PoolInitialized {
```

Comment:
I think we should co-relate the pools to their status via their name

Reply:
Yeah will change that..

Reply:
actually, we do not need that @dvaldivia because we are only interested in knowing exactly which pool is initialized here - we know that any pool that is initialized can be restarted to cater for adding a new pool.